pkg/storage/backend-storage/backend-storage.go

```diff
@@ -118,3 +119,38 @@ func CreateIfNeeded(vmi *corev1.VirtualMachineInstance, clusterConfig *virtconfi
 
 	return err
 }
+
+// IsPVCReady returns true if either:
+// - No PVC is needed for the VMI since it doesn't use backend storage
+// - The backend storage PVC is bound
+// - The backend storage PVC is pending uses a WaitForFirstCustomer storage class
+func IsPVCReady(vmi *corev1.VirtualMachineInstance, client kubecli.KubevirtClient) (bool, error) {
+	if !IsBackendStorageNeededForVMI(&vmi.Spec) {
+		return true, nil
+	}
+
+	pvc, err := client.CoreV1().PersistentVolumeClaims(vmi.Namespace).Get(context.Background(), PVCForVMI(vmi), metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	switch pvc.Status.Phase {
+	case v1.ClaimBound:
+		return true, nil
+	case v1.ClaimLost:
+		return false, fmt.Errorf("backend storage PVC lost")
+	case v1.ClaimPending:
+		if pvc.Spec.StorageClassName == nil {
+			return false, fmt.Errorf("no storage class name")
+		}
+		sc, err := client.StorageV1().StorageClasses().Get(context.Background(), *pvc.Spec.StorageClassName, metav1.GetOptions{})
```

Comment:
`VirtControllerApp` has a storageclass informer you can use

Reply:
Done in new commit, thanks!